### PR TITLE
Feture 6 add service

### PR DESCRIPTION
**Pull request by alesyalis**
#6 - converted the controller into Task, add CourseService, add mapperConfig, registration Courseprofile, add model FeedbackType and fild into Feedback;
#7  - Mongo add id UUID

### CODE REVIEW

File: Recrutify/Recrutify.Host/Controllers/CourseController.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        public async Task<ActionResult<List<Course>>> GetAsync()
+        {
+            var result = await _courseService.GetAllAsync();
+            return Ok(result);
+        }
+
+        [HttpPost]
+        public async Task<ActionResult> AddCourse(CourseDto courseDto)
+        {
+            await _courseService.CreatAsynk(courseDto);
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Controllers/CourseController.cs:31]: CreateAsyn**c**

File: Recrutify/Recrutify.DataAccess/Repositories/Abstract/ICourseRepository.cs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ﻿namespace Recrutify.DataAccess.Repositories.Abstract
 {
-    public interface ICourseRepository : IBaseRepository<Course>
+    public interface ICourseRepository : IBaseRepository<Project>
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/Abstract/ICourseRepository.cs:3]: rename to IProjectRepository

File: Recrutify/Recrutify.DataAccess/Repositories/CourseRepository.cs [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 namespace Recrutify.DataAccess.Repositories
 {
-    public class CourseRepository : BaseRepository<Course>, ICourseRepository
+    public class CourseRepository : BaseRepository<Project>, ICourseRepository
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.DataAccess/Repositories/CourseRepository.cs:7]: rename to ProjectRepository

File: Recrutify/Recrutify.Host/Configuration/Profiles/CourseProfile.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace Recrutify.Host.Configuration.Profiles
+{
+    public class CourseProfile : AutoMapper.Profile
+    {
+        public CourseProfile()
+        {
+            CreateMap<CreateCourseDto, Project>()
+                .ForMember(dest => dest.Id, conf => conf.MapFrom(src => Guid.NewGuid()));
+            CreateMap<Project, CourseDto>();
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Configuration/Profiles/CourseProfile.cs:13]: Rename to ProjectDTO

File: Recrutify/Recrutify.Services/Dtos/CourseDto.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using Recrutify.DataAccess.Models;
+
+namespace Recrutify.Services.Dtos
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Dtos/CourseDto.cs:5]: rename folder to DTOs

File: Recrutify/Recrutify.Host/Configuration/Profiles/CourseProfile.cs [hunk 6]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using Recrutify.DataAccess;
+using Recrutify.Services.Dtos;
+
+namespace Recrutify.Host.Configuration.Profiles
+{
+    public class CourseProfile : AutoMapper.Profile
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Configuration/Profiles/CourseProfile.cs:7]: rename to ProjectProfile

File: Recrutify/Recrutify.Host/Controllers/CourseController.cs (hunk 1 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Host/Controllers/CourseController.cs]: ready

File: Recrutify/Recrutify.Host/Startup.cs [hunk 7]
```diff
@@ -20,10 +27,15 @@ public Startup(IConfiguration configuration)
...
         public void ConfigureServices(IServiceCollection services)
         {
+            MongoDefaults.GuidRepresentation = MongoDB.Bson.GuidRepresentation.Standard;
             services.Configure<MongoSettings>(
                 Configuration.GetSection(nameof(MongoSettings)));
-
+            services.AddSingleton<ICourseRepository, CourseRepository>();
+            services.AddSingleton<ICourseService, CourseService>();
+            var mapper = MapperConfig.GetConfiguration().CreateMapper();
+            services.AddSingleton(mapper);
```
[the remark this answers is not in the document]
Comment (alesyalis) [reply Recrutify/Recrutify.Host/Startup.cs:41]: ready

File: Recrutify/Recrutify.Services/Servises/CourseService.cs [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        private readonly ICourseRepository _courseRepository;
+        private readonly IMapper _mapper;
+
+        public CourseService(ICourseRepository courseRepository, IMapper mapper)
+        {
+            _courseRepository = courseRepository;
+            _mapper = mapper;
+        }
+
+        public async Task CreatAsync(CourseDto courseDto)
```
[the remark this answers is not in the document]
Comment (alesyalis) [reply Recrutify/Recrutify.Services/Servises/CourseService.cs]: ready

File: Recrutify/Recrutify.Host/Configuration/Profiles/CourseProfile.cs [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    {
+        public CourseProfile()
+        {
+            CreateMap<CreateCourseDto, Project>()
```
[the remark this answers is not in the document]
Comment (alesyalis) [reply Recrutify/Recrutify.Host/Configuration/Profiles/CourseProfile.cs:11]: ready

File: Recrutify/Recrutify.Host/Configuration/Profiles/CourseProfile.cs (hunk 4 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Host/Configuration/Profiles/CourseProfile.cs:13]: ready

File: Recrutify/Recrutify.Services/Servises/Abstract/ICourseService.cs [hunk 10]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Recrutify.Services.Dtos;
+
+namespace Recrutify.Services.Servises.Abstract
+{
+    public interface ICourseService
```
[the remark this answers is not in the document]
Comment (alesyalis) [reply Recrutify/Recrutify.Services/Servises/Abstract/ICourseService.cs:7]: ready

File: Recrutify/Recrutify.Host/Configuration/Profiles/CourseProfile.cs (hunk 6 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Host/Configuration/Profiles/CourseProfile.cs:7]: ready

File: Recrutify/Recrutify.Host/Controllers/CourseController.cs [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        private readonly IProjectService _courseService;
+
+        public CourseController(IProjectService courseService)
+        {
+            _courseService = courseService;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<List<Project>>> GetAsync()
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Controllers/CourseController.cs:22]: return type should be Task<ActionResult<List<ProjectDTO>>>

File: Recrutify/Recrutify.Host/Controllers/CourseController.cs [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+        {
+            var result = await _courseService.GetAllAsync();
+            return Ok(result);
+        }
+
+        [HttpPost]
+        public async Task<ActionResult> AddCourse(ProjectCreateDTO courseDto)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Host/Controllers/CourseController.cs:29]: return type should be Task<ActionResult<ProjectDTO>>

**Review by elizavetapavliv — APPROVED** [review]

**Pull request merged**